bids/cli.py

```diff
@@ -0,0 +1,123 @@
+
+
+class PathOrRegex(click.ParamType):
+    "A helper Type to parse BIDSLayoutIndexer ignore/force entries"
+    name = "path or /regex/"
+
+    def convert(self, value, param, ctx):
+        if value.startswith('/') and value.endswith('/'):
+            # regex pattern
+            import re
+            value = re.compile(value[1:-1])
+        # otherwise, return as is
+        return value
+
+
+# create group of commands as entrypoint
+@click.group(context_settings=CONTEXT_SETTINGS)
+@click.version_option(__version__, prog_name='pybids')
+def cli():
+    """Command-line interface for PyBIDS operations"""
+    pass
+
+
+@cli.command(context_settings=CONTEXT_SETTINGS)
+@click.argument('root', type=click.Path(file_okay=False, exists=True))
+@click.option('--db-path', type=click.Path(file_okay=False, resolve_path=True),
+              help="Path to save database index.")
+@click.option('--derivatives', multiple=True, default=False, show_default=True,
+              help="Specifies whether and/or which derivatives to to index.")
+@click.option('--reset-db', default=False, show_default=True, is_flag=True,
+              help="Remove existing database index if present.")
+@click.option('--validate/--no-validate', default=True, show_default=True,
+              help="Check for BIDS compliance when indexing files.")
+@click.option('--config', multiple=True,
+              help="Optional name(s) of configuration file(s) to use.")
+@click.option('--index-metadata/--no-index-metadata', default=False, show_default=True,
+              help="Include metadata when indexing files.")
+@click.option('--ignore', multiple=True, type=PathOrRegex(),
+              help="Path (from root) or regex to exclude from indexing. "
+                   "Regex entries need to fitted with leading and trailing '/'.")
+@click.option('--force-index', multiple=True, type=PathOrRegex(),
+              help="Path (from root) or regex to include when indexing. "
+                   "Regex entries need to fitted with leading and trailing '/'.")
+@click.option('--config-filename', type=click.Path(),
+              default="layout_config.json", show_default=True,
+              help="Name of filename within directories that contains configuration information.")
+def layout(
+    root,
+    db_path,
+    derivatives,
+    reset_db,
+    validate,
+    config,
+    index_metadata,
+    ignore,
+    force_index,
+    config_filename,
+):
+    """
+    Initialize a BIDSLayout.
+
+    If ``--db-path`` is provided, an SQLite database index for this BIDS dataset will be created.
+
+    """
+    # ensure empty multiples are set to None
+    derivatives = _validate_multiple(derivatives, retval=False)
+    config = _validate_multiple(config)
+    ignore = _validate_multiple(ignore)
+    force_index = _validate_multiple(force_index)
+
+    if db_path and not (Path(db_path) / 'layout_index.sqlite').exists():
+        reset_db = True
+
+    indexer = _init_indexer(
```

Comment:
I don't understand why this isn't just `BIDSLayoutIndexer`.

Reply:
The idea was for `_init_indexer` and `_init_layout` to be potentially reusable across other subcommands. But I'll just leave it for a future refactor, if necessary.